src/main/java/dan200/computercraft/shared/turtle/upgrades/TurtleShovel.java

```diff
@@ -41,4 +51,20 @@ protected boolean canBreakBlock( World world, BlockPos pos )
         }
         return false;
     }
+
+    @Nonnull
+    @Override
+    public TurtleCommandResult useTool( @Nonnull ITurtleAccess turtle, @Nonnull TurtleSide side, @Nonnull TurtleVerb verb, @Nonnull EnumFacing direction )
+    {
+        if( verb == TurtleVerb.Dig )
+        {
+            ItemStack shovel = m_item.copy();
+            ItemStack remainder = TurtlePlaceCommand.deploy( shovel, turtle, direction, null, null );
+            if( remainder != shovel )
```

Comment:
Hrrmrm. It _should_ be, as the right click action for hoes/shovels should return the original stack. However it is possible that could change in the future. Would you like me to change `TurtleHoe`'s implementation as well?